fetcher/tar.py

```diff
@@ -99,18 +101,26 @@ def unpack(self, downloadFirst=False):
         update = tarfile.open(name=self.workspacefile, mode='r:*')
         update.extractall(self.workspaceunpack)
 
-        # Save the rootfs filename of easy access
-        rootfsfiles = os.listdir(self.workspace + "/update/resin-root")
-        self.rootfsarchive = self.workspace + "/update/resin-root/" + os.listdir(self.workspace + "/update/resin-root")[0]
-
         log.debug("Unpacked " + self.workspacefile + " in " + self.workspaceunpack)
         return True
 
+    def unpackQuirks(self, location):
+        '''Copies the content of <quirks> directory from an update to <location>.'''
+        quirks_path = os.path.join(self.workspaceunpack, 'quirks')
+        log.info("Unpacking rootfs quirks from " + quirks_path + " to " + self.workspaceunpack + ".")
+        if not os.path.isdir(quirks_path):
+            log.debug("No quirks found. Skipping.")
+            return True
+        if not safeCopy(quirks_path, location, sync=False):
+            log.error("Failed to unpack quirks.")
+            return False
+        log.info("Unpack rootfs quirks done.")
+        return True
+
     def unpackRootfs(self, location):
         log.info("Unpack rootfs started... this can take a couple of seconds or even minutes...")
-        update = tarfile.open(self.rootfsarchive)
-        update.extractall(location)
-        log.debug("Unpacked rootfs " + self.rootfsarchive + " in " + location)
+        safeCopy(self.workspaceunpack, location, sync=False, ignore=['resin-boot', 'quirks'])
```

Comment:
Why don't need the ignore mechanism at all if the operations happen in the right order. Can we only unpack __once__ into the inactive root partition and then do the two copy operations for `resin-boot` and `quirks` from there?